core/src/main/scala/com/yahoo/maha/core/FilterOperation.scala

```diff
@@ -1578,7 +1579,8 @@ object FilterSql {
                    columnsByNameMap: Map[String, Column],
                    engine: Engine,
                    literalMapper: SqlLiteralMapper,
-                   grainOption: Option[Grain] = None): SqlResult = {
+                   grainOption: Option[Grain] = None,
+                   request: Option[ReportingRequest] = None): SqlResult = {
```

Comment:
you could directly pass ReportingRequest as it is guaranteed to be valid in filter render context

Reply:
thanks, updated the PR and passing the reportingRequest directly